deployer/machine.go

```diff
@@ -14,6 +14,7 @@ func StateMachine() (*machine.StateMachine, error) {
     "States": {
       "Validate": {
         "Type": "TaskFn",
+        "Resource": "arn:aws:lambda:{{aws_region}}:{{aws_account}}:function:{{lambda_name}}",
```

Comment:
I think my biggest nit is that all the `Resources` must have the same aws_region, aws_account, and lambda_name.

That is quite a tight restriction - what if you want to use different lambdas in each step?

Reply:
An idea to have to get around this is to use a built in templating tool that golang has - a quick search shows https://golang.org/pkg/text/template/

Maybe you could just let users specify in the machine $resource_arn_1, $resource_arn_2, or something of the sort.

This would also cut out the to.InterpolateArnVariables and its tests, which is nice.

Reply:
then you have to hardcode the lambda name for now.